src/main/java/com/google/sps/servlets/DashboardServlet.java

```diff
@@ -14,62 +14,125 @@
 import com.google.appengine.api.datastore.Query.FilterPredicate;
 import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
 import com.google.sps.servlets.Property;
 import java.lang.reflect.Field; 
 import java.util.HashMap;
+import com.google.sps.servlets.RequestUtils;
 
 @WebServlet("/dashboard")
 public class DashboardServlet extends HttpServlet {
 
+  private final long MILLISECONDS_IN_DAY = 86400000;
+  private final String DEFAULT_START_TIMESTAMP = "0";
+  private final String DEFAULT_END_TIMESTAMP = String.valueOf(System.currentTimeMillis());
+  private String startTimestamp;
+  private String endTimestamp;
+
  /**
   * This route will return a JSON with the percentages of each interaction
   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    String requestStartTimestamp = RequestUtils.getParameter(request, "startTime");
+    String requestEndTimestamp = RequestUtils.getParameter(request, "endTime");
+
+    validateAndSetVariables(requestStartTimestamp, requestEndTimestamp);    
+    
     try {
-      HashMap<String, Double> interactionPercentages = calculatePercentages();
+      HashMap<String, Double> dataToSend = calculatePercentages();
       
-      String jsonToSend = convertToJson(interactionPercentages);
+      String jsonToSend = convertToJson(dataToSend);
 
+      response.addHeader("Access-Control-Allow-Origin", "*");
+      response.setStatus(HttpServletResponse.SC_OK);
       response.setContentType("application/json; charset=UTF-8");
       response.getWriter().println(jsonToSend);
     
     } catch (IllegalAccessException exception) {
+
       response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
     }
   }
 
  /**
-  * This function queries for and calculates the percentages of each interaction.
+  * This function queries for and calculates the percentages of each
+  * interaction.
+  *
+  * @param startTimestamp Timestamp for the beginning of the date range that is requested.
+  * @param endTimestamp Timestamp for the end of the date range that is requested.
   * @return A hashmap with the interactions as keys and the percentages as values.
   */
-  public HashMap<String, Double> calculatePercentages() throws IllegalAccessException {
+  private HashMap<String, Double> calculatePercentages() throws IllegalAccessException {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery interactions = datastore.prepare(new Query(DBUtilities.INTERACTION_TABLE));
+
+    final Filter startTimestampFilter =  new FilterPredicate(Property.TIMESTAMP, FilterOperator.GREATER_THAN_OR_EQUAL, this.startTimestamp);
+    final Filter endTimestampFilter =  new FilterPredicate(Property.TIMESTAMP, FilterOperator.LESS_THAN_OR_EQUAL, this.endTimestamp);
+    
+    // combine the two filters with the 'and' operator 
```

Comment:
This comment is just repeating what the code says, imo its not adding any value

Reply:
removed